repo/branson/package.py

```diff
@@ -39,16 +44,39 @@ class Branson(CMakePackage):
     depends_on(Boost.with_default_variants, when="@:0.81")
     depends_on("metis")
     depends_on("parmetis", when="@:0.81")
+    depends_on("caliper", when="+caliper")
+    depends_on("adiak", when="+caliper")
 
     root_cmakelists_dir = "src"
 
+    def patch(self):
+        ppu_intrinsics_file = os.path.join(self.stage.source_path, "src", "random123", "features", "ppu_intrinsics.h")
+        with open(ppu_intrinsics_file , "w") as f:
+            pass
+
+    def setup_build_environment(self, env):
+        if not self.spec.satisfies("+cuda"):
+            env.unset("CUDA_HOME")
+            env.unset("CUDADIR")
+            env.unset("CUDACXX")
+
     def cmake_args(self):
         spec = self.spec
         args = []
         #args.append("--enable-mpi")
         args.append(f"-DCMAKE_C_COMPILER={spec['mpi'].mpicc}")
         args.append(f"-DCMAKE_CXX_COMPILER={spec['mpi'].mpicxx}")
         args.append(f"-DCMAKE_Fortran_COMPILER={spec['mpi'].mpifc}")
+
+        cflags = " ".join(self.compiler.flags['cflags']) if 'cflags' in self.compiler.flags else ""
+        cxxflags = " ".join(self.compiler.flags['cxxflags']) if 'cxxflags' in self.compiler.flags else ""
+
+        args.append("-DCMAKE_C_FLAGS={} -I{}/src/random123/features".format(cflags, self.stage.source_path))
+        args.append("-DCMAKE_CXX_FLAGS={} -I{}/src/random123/features".format(cxxflags, self.stage.source_path))
+        args.append(f"-DBUILD_TESTING=OFF")
```

Comment:
@pearce8 This is just a temporary fix for lasssen. Need to work with the branson team to fix build issues on that system.